src/MSBuildLocator/VisualStudioInstanceQueryOptions.cs

```diff
@@ -33,5 +33,11 @@ public class VisualStudioInstanceQueryOptions
         ///     Working directory to use when querying for instances. Ensure it is the project directory to pick up the right global.json.
         /// </summary>
         public string WorkingDirectory { get; set; } = Environment.CurrentDirectory;
+
+        /// <summary>
+        ///     This variable enables the removal of the existing restriction on querying installed Visual Studio instances 
+        ///     with a runtime version lower than or equal to the one in the current environment.
+        /// </summary>
+        public bool AllowQueryAllRuntimeVersions { get; set; } = Environment.GetEnvironmentVariable("DOTNET_MSBUILD_QUERY_ALL_RUNTIMES") == "1";
```

Comment:
What's motivating the environment variable here, instead of just requiring this setting?

Reply:
I wanted to avoid extending public API. 
If adding a flag is a better option I can change.

Reply:
fixed